### PR TITLE
Split IOperationTests into one part per type

### PR DESCRIPTION
**Pull request by sharwell**
Fixes [AB#1350975](https://devdiv.visualstudio.com/0bdbc590-a062-4c3f-b0f6-9383f67865ee/_workitems/edit/1350975)

### CODE REVIEW

**Comment by 333fred** [comment]
While I'm fine with this change in general, I'm not fine with considering it a fix for that bug. Yes, it's a worst-case scenario. But given that there were literally no nullable suppressions in syntax in the file I was editing, I don't think that it's OK to not have a fast-path.

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_ArrayCreationAndInitializer.cs [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
 namespace Microsoft.CodeAnalysis.CSharp.UnitTests
 {
-    public partial class IOperationTests : SemanticModelTestBase
+    public class IOperationTests_ArrayCreationAndInitializer : SemanticModelTestBase
```
Comment (333fred) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_ArrayCreationAndInitializer.cs:14]: If it's not too much work, I'd _prefer_ if we named these things something like `ArrayCreationAndInitializerOperationTests`.

**Comment by sharwell** [comment]
I filed https://github.com/dotnet/roslyn/issues/54577 to address the underlying issue.

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_ArrayCreationAndInitializer.cs (hunk 1 above)
Comment (sharwell) [reply src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_ArrayCreationAndInitializer.cs:14]: The file can be renamed in a follow-up.

Comment (333fred) [reply src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_ArrayCreationAndInitializer.cs:14]: Alright. That's why I left it at prefer.

**Comment by 333fred** [comment]
> I filed #54577 to address the underlying issue.

Thanks.

**Review by 333fred — APPROVED** [review]

**Review by jcouv — APPROVED** [review]
LGTM Thanks (iteration 1)

**Pull request merged**